### PR TITLE
Fix segmentation fault when using INT4 with CONCAT MLP

### PR DESCRIPTION
**Pull request by abenmao**

### CODE REVIEW

File: src/layers/mlp_chatglm2.h [hunk 1]
```diff
@@ -39,7 +39,7 @@ class ChatGLM2MLP : public LlamaMLP<WeiT> {
         int colSplit = range.second - range.first;
 
         setMLPOPTConfig();
-        if (!enableCATMLP) {
+        if (!enableCATMLP or std::is_same_v<WeiT, uint4x2_t>) {
```
Comment (changqi1) [inline src/layers/mlp_chatglm2.h:42]: nf4x2_t?

**Comment by changqi1** [comment]
@abenmao  All pass?

**Review by pujiang2018 — APPROVED** [review]

File: src/layers/mlp_chatglm2.h (hunk 1 above)
Comment (abenmao) [reply src/layers/mlp_chatglm2.h]: done~

**Comment by abenmao** [comment]
> @abenmao All pass?

yes,all pass~

**Pull request merged**